bigquery-connector-common/src/main/java/com/google/cloud/bigquery/connector/common/BigQueryClient.java

```diff
@@ -616,4 +631,30 @@ Job waitForJob(Job job) {
       }
     }
   }
+
+  static class JobConfigurationFactory {
+    private final ImmutableMap<String, String> labels;
+
+    public JobConfigurationFactory(Map<String, String> labels) {
+      this.labels = ImmutableMap.copyOf(labels);
+    }
+
+    QueryJobConfiguration.Builder createQueryJobConfigurationBuilder(String querySql) {
+      QueryJobConfiguration.Builder builder = QueryJobConfiguration.newBuilder(querySql);
+      if (labels != null && !labels.isEmpty()) {
+        builder.setLabels(labels);
+      }
+      return builder;
+    }
+
+    LoadJobConfiguration.Builder createLoaJobConfigurationBuilder(
```

Comment:
Typo: Should be `createLoadJobConfigurationBuilder` and not `createLoaJobConfigurationBuilder`

Reply:
Also, is this method not used anywhere?

Reply:
Fixed, also fixed the usage in loadDataIntoTable()